pkg/config/analysis/analyzers/util/in_mesh.go

```diff
@@ -27,23 +29,30 @@ import (
 // DeploymentinMesh returns true if deployment is in the service mesh (has sidecar)
 func DeploymentInMesh(r *resource.Instance, c analysis.Context) bool {
 	d := r.Message.(*apps_v1.DeploymentSpec)
-	return inMesh(d.Template.Annotations, resource.Namespace(r.Metadata.FullName.Namespace.String()), d.Template.Spec.Containers, c)
+	return inMesh(d.Template.Annotations, d.Template.Labels,
+		resource.Namespace(r.Metadata.FullName.Namespace.String()), d.Template.Spec.Containers, c)
 }
 
 // PodInMesh returns true if a Pod is in the service mesh (has sidecar)
 func PodInMesh(r *resource.Instance, c analysis.Context) bool {
 	p := r.Message.(*v1.PodSpec)
-	return inMesh(r.Metadata.Annotations, r.Metadata.FullName.Namespace, p.Containers, c)
+	return inMesh(r.Metadata.Annotations, r.Metadata.Labels,
+		r.Metadata.FullName.Namespace, p.Containers, c)
 }
 
-func inMesh(annos map[string]string, namespace resource.Namespace, containers []v1.Container, c analysis.Context) bool {
+func inMesh(annos, labels map[string]string, namespace resource.Namespace, containers []v1.Container, c analysis.Context) bool {
 	// If pod has the sidecar container set, then, the pod is in the mesh
 	if hasIstioProxy(containers) {
 		return true
 	}
 
+	// If Pod has labels, return the injection label value
```

Comment:
The check on L55 is more robust

Reply:
Confused, shouldn't labels be respected over annotations

Reply:
I misunderstood